client-app/src/pages/account/views/AddUserModal.vue

```diff
@@ -0,0 +1,241 @@
+<template>
+  <b-modal id="addUserModal"
+           no-close-on-backdrop
+           @hide="resetForm">
+    <div slot="modal-title">
+      {{ $t("account.users.add-user.title") }}
+    </div>
+    <b-form>
+      <b-form-group :label="$t('account.users.add-user.email-label')"
+                    label-for="email">
+        <b-form-input
+          id="email"
+          v-model="$v.user.email.$model"
+          type="email"
+          :class="{ 'is-invalid': $v.user.email.$error }"
+          :placeholder="$t('account.users.add-user.email-placeholder')"
+          @blur="$v.user.email.$touch()"></b-form-input>
+        <b-form-invalid-feedback v-if="!$v.user.email.email">
+          {{ $t("account.users.add-user.email-not-valid") }}
+        </b-form-invalid-feedback>
+        <b-form-invalid-feedback v-if="!$v.user.email.required">
+          {{ $t("account.users.add-user.email-required") }}
+        </b-form-invalid-feedback>
+      </b-form-group>
+
+      <b-form-group
+        label-for="firstName">
+        <template v-slot:label>
+          {{ $t("account.users.add-user.first-name-label") }} - <span class="font-italic">{{ $t("account.users.add-user.optional") }}</span>
+        </template>
+        <b-form-input
+          id="firstName"
+          v-model="user.firstName"
+          type="text"
+          :placeholder="$t('account.users.add-user.first-name-placeholder')"></b-form-input>
+      </b-form-group>
+
+      <b-form-group
+        label-for="lastName">
+        <template v-slot:label>
+          {{ $t("account.users.add-user.last-name-label") }} - <span class="font-italic">{{ $t("account.users.add-user.optional") }}</span>
+        </template>
+        <b-form-input
+          id="lastName"
+          v-model="user.lastName"
+          type="text"
+          :placeholder="$t('account.users.add-user.last-name-placeholder')"></b-form-input>
+      </b-form-group>
+
+      <b-form-group :label="$t('account.users.add-user.username-label')"
+                    label-for="userName">
+        <b-form-input
+          id="userName"
+          v-model="$v.user.userName.$model"
+          :class="{ 'is-invalid': $v.user.userName.$error }"
+          type="text"
+          :placeholder="$t('account.users.add-user.username-placeholder')"
+          aria-describedby="userName-help-block"
+          @blur="$v.user.userName.$touch()"></b-form-input>
+        <b-form-invalid-feedback v-if="!$v.user.userName.required">
+          {{ $t("account.users.add-user.username-required") }}
+        </b-form-invalid-feedback>
+      </b-form-group>
+
+      <b-form-group :label="$t('account.users.add-user.role-label')"
+                    label-for="role">
+        <b-form-select
+          id="role"
+          v-model="$v.user.role.$model"
+          :class="{ 'is-invalid': $v.user.role.$error }"
+          aria-describedby="role-help-block"
+          plain
+          @blur.native="$v.user.role.$touch()">
```

Comment:
Would it be better to use component events?
https://bootstrap-vue.js.org/docs/components/form-select#comp-ref-b-form-select-events